internal-dns/src/config.rs

```diff
@@ -60,60 +60,32 @@
 //!
 //! This module provides types used to assemble that configuration.
 
-use crate::names::{BackendName, ServiceName, DNS_ZONE};
+use crate::names::{ServiceName, DNS_ZONE};
 use anyhow::{anyhow, ensure};
 use dns_service_client::types::{DnsConfigParams, DnsConfigZone, DnsRecord};
 use std::collections::BTreeMap;
 use std::net::Ipv6Addr;
 use uuid::Uuid;
 
-/// Describes the DNS name that will be used for a control plane service
-///
-/// This does not describe the SRV record itself.
-#[derive(Clone, Debug, Hash, Eq, Ord, PartialEq, PartialOrd)]
-pub enum SRV {
-    /// A service identified and accessed by name, such as "nexus", "CRDB", etc.
-    ///
-    /// This is used in cases where services are interchangeable.
-    Service(ServiceName),
-
-    /// A service identified by name and a unique identifier.
-    ///
-    /// This is used in cases where services are not interchangeable, such as
-    /// for the Sled agent.
-    Backend(BackendName, Uuid),
-}
-
-impl SRV {
-    /// Returns the DNS name for this service, ignoring the zone part of the DNS
-    /// name
-    pub(crate) fn dns_name(&self) -> String {
-        match &self {
-            SRV::Service(name) => format!("_{}._tcp", name),
-            SRV::Backend(name, id) => format!("_{}._tcp.{}", name, id),
-        }
-    }
-}
-
 /// Describes the DNS name used for a control plane host
 ///
 /// This does not describe the AAA record itself.
 #[derive(Clone, Debug, PartialEq, PartialOrd)]
-enum AAAA {
+enum Host {
     /// Identifies an AAAA record for a sled.
     Sled(Uuid),
 
     /// Identifies an AAAA record for a zone within a sled.
     Zone(Uuid),
```

Comment:
Should we update the commentary here? "Identifies an AAAA record" seems a little odd now.